test/integration-test/src/utils.rs

```diff
@@ -0,0 +1,76 @@
+//! Utilities to run tests
+
+use std::{
+    fs::File,
+    io::Write,
+    os::fd::AsRawFd,
+    process::{Command, Stdio},
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use libc::{setns, CLONE_NEWNET};
+
+fn script(script: &str) {
+    // We pipe stdout/err to capture it and print it ourselves, so that it gets captured by the
+    // test harness
+    let mut bash = Command::new("/bin/bash")
+        .arg("-e")
+        .stdin(Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    // Send script
+    let mut stdin = bash.stdin.take().unwrap();
+    stdin.write_all(script.as_bytes()).unwrap();
+    stdin.write_all(b"\n").unwrap();
+    drop(stdin);
+
+    // Wait for it to finish
+    let status = bash.wait().unwrap();
+
+    assert_eq!(status.code(), Some(0));
+}
+
+pub struct Netns {
+    pub name: String,
+    pub fd: File,
+}
+
+impl Netns {
+    pub fn new() -> Self {
+        static COUNTER: AtomicU64 = AtomicU64::new(0);
```

Comment:
Would it make sense to move `COUNTER` outside of Netns?
Wondering if we ever run in parallel whether 2 tests could pick netns 0 at the same time.

Reply:
I don't follow you -- this is a static atomic, so as long as it's the same process, no two test will pick the same number. And since those integration tests are unit tests in the cargo sense, that's what happens.

Moving it outside of netns (out of `new` for that matter) would only make it accessible to foreign code (respecting visibility ofc)

Reply:
we might want to include the pid of the current process in the netns name to avoid concurrent processes from picking the same one.